examples/RPI/Example1_Basics/Example1_Basics.cpp

```diff
@@ -0,0 +1,67 @@
+/****************************************************************
+ * Example1_Basics.cpp
+ * HSCDTD008A Library Demo
+ * Tilman Glötzner
+ * Original Creation Date: 2022-09-03
+ * Heavily based on the code of Bob Veringa
+ *
+ * Distributed as-is; no warranty is given.
+ ***************************************************************/
+
+#include "hscdtd008a.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <math.h>
+
+// Create an instance of the sensor.
+HSCDTD008A geomag;
+
+
+void setup() {
+  hscdtd_status_t status;
+
+
+  geomag.begin();
+  // If you know the I2C address is different than in the provided
+  // data sheet. Uncomment the line below, and configure the address.
+  geomag.begin(0x0F);
+
+  // Initialize the hardware.
+  status = geomag.initialize();
+  if (status != HSCDTD_STAT_OK) {
+    printf("Failed to initialize sensor. Status:%d. Check wiring.\n",status);
+
+    // Halt program here.
+    exit(1);
+  }
+  // Compensate for temperature.
+  geomag.temperatureCompensation();
+
+}
+
+void loop() {
+  hscdtd_status_t status;
+
+  // Explicitly start a reading.
+  status = geomag.startMeasurement();
+  // If the status is OK then we can print the result.
+  if (status == HSCDTD_STAT_OK) {
+    float Babs = sqrt(pow(geomag.mag.mag_x,2) + pow(geomag.mag.mag_y,2) + pow(geomag.mag.mag_z,2));
```

Comment:
Line 50 (or 51 in my suggestion) is over 80 characters long. There is not really a clear way to break up this bit of code, maybe something like this could work? But I am uncertain if all compilers support this.

```c
    bAbs = sqrt(
      pow(geomag.mag.mag_x, 2) 
      + pow(geomag.mag.mag_y, 2) 
      + pow(geomag.mag.mag_z, 2)
    );
```

Another option would be to split out sub products. But this would change the function a bit. To something like this
```c
  hscdtd_status_t status;
  // Explicitly start a reading.
  status = geomag.startMeasurement();
  // If the status is OK then we can print the result.
  if (status == HSCDTD_STAT_OK) {
    float bAbs;
    float subX = pow(geomag.mag.mag_x, 2);
    float subY = pow(geomag.mag.mag_y, 2);
    float subZ = pow(geomag.mag.mag_z, 2);
    bAbs = sqrt(subX + subY + subZ);

    printf("X: %f uT,\tY: %f uT,\tZ: %f uT;\t|B|: %f uT\n",
    	geomag.mag.mag_x, geomag.mag.mag_y, geomag.mag.mag_z, bAbs);
```

This would need some better naming, but since I am not sure what those components do, I have just added temporary naming here.

Reply:
The gcc compiler can handle code spanning over several lines. I will adapt it as you suggested, i.e. break up the line in several line connected by '+'.